### PR TITLE
Add an E2E GKE test for secure image

### PR DESCRIPTION
**Pull request by qiwzhang**

### CODE REVIEW

File: script/gke-e2e.sh [hunk 1]
```diff
@@ -0,0 +1,113 @@
+#!/bin/bash
```
Comment (JLXIA) [inline script/gke-e2e.sh]: Why adding a new gke-e2e.sh, while lots of common code with e2e-kube.sh?

Comment (qiwzhang) [reply script/gke-e2e.sh]: e2e-kube is using espcli.   I want to move away  from espcli.

That is why I create a new script to use the deployment yaml file directly.

for new gke tests, they will be using this new script.

File: test/bookstore/gke/deploy_secure_template.yaml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      # [END esp]
+        ports:
+        - containerPort: 8080
+          protocol: TCP
+        - containerPort: 8090
+          protocol: TCP
+        securityContext:
+          readOnlyRootFilesystem: true
+        volumeMounts:
+          - mountPath: /var/log/nginx
```
Comment (inklesspen) [inline test/bookstore/gke/deploy_secure_template.yaml]: knative is going to mount a volume at `/var/log`, not `/var/log/nginx`. Your test should probably do the same.

Comment (qiwzhang) [reply test/bookstore/gke/deploy_secure_template.yaml]: ok , will change to /var/log

**Pull request merged**